webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/components/ShowNodeDetailsFromNode.scala

```diff
@@ -195,23 +195,27 @@ class ShowNodeDetailsFromNode(
   }
 
   def mainDispatch = Map(
-    "popupDetails"    -> { _:NodeSeq => privateDisplay(true, false)  }
-  , "popupCompliance" -> { _:NodeSeq => privateDisplay(true, true)   }
-  , "mainDetails"     -> { _:NodeSeq => privateDisplay(false, false) }
-  , "mainCompliance"  -> { _:NodeSeq => privateDisplay(false, true)  }
+    "popupDetails"    -> { _:NodeSeq => privateDisplay(true, false, false)  }
+  , "popupCompliance" -> { _:NodeSeq => privateDisplay(true, true, false)   }
+  , "popupSystem"     -> { _:NodeSeq => privateDisplay(true, true, true)   }
+  , "mainDetails"     -> { _:NodeSeq => privateDisplay(false, false, false) }
+  , "mainCompliance"  -> { _:NodeSeq => privateDisplay(false, true, false)  }
+  , "mainSystem"      -> { _:NodeSeq => privateDisplay(false, false, true)  }
   )
 
-  def display(popupDisplay : Boolean, complianceDisplay : Boolean) : NodeSeq = {
-    val dispatchName = (popupDisplay, complianceDisplay) match {
-      case (true, true)  => "popupCompliance"
-      case (true, _)     => "popupDetails"
-      case (false, true) => "mainCompliance"
-      case (false, _)    => "mainDetails"
+  def display(popupDisplay : Boolean, complianceDisplay : Boolean, systemStatus : Boolean) : NodeSeq = {
+    val dispatchName = (popupDisplay, complianceDisplay, systemStatus) match {
+      case (true, _, true)  => "popupSystem"
```

Comment:
trying to rebuild what case we are from a partial truth table seems fichsy. Is it possible to just use an enumeration so that the caller just say `display(PopupCompliance)` etc ?